source/extensions/filters/http/health_check/health_check.cc

```diff
@@ -173,9 +174,16 @@ void HealthCheckFilter::onComplete() {
     if (!Http::CodeUtility::is2xx(enumToInt(final_status))) {
       callbacks_->streamInfo().setResponseFlag(
           StreamInfo::CoreResponseFlag::FailedLocalHealthCheck);
+      stats_->failed_.inc();
```

Comment:
Thinking out loud: should the failed_/ok_ also be incremented for the cached response?
Either way, the doc above (about the new statistics) should reflect this.

Reply:
I think it does make sense to have fail/ok include the cached response as well. I have clarified this in the doc. Let me know if you think otherwise.